sympy/physics/control/lti.py

```diff
@@ -802,80 +918,72 @@ def doit(self, **kwargs):
         TransferFunction((2 - s**3)*(-a*p**2 - b*s), (-p + s)*(s**4 + 5*s + 6), s)
 
         """
-        res = None
-        for arg in self.args:
-            arg = arg.doit()
-            if res is None:
-                res = arg
-            else:
-                num_ = arg.num * res.num
-                den_ = arg.den * res.den
-                res = TransferFunction(num_, den_, self.var)
-        return res
+        def _SISO_doit():
+            _arg = (arg.doit().to_expr() for arg in self.args)
+            res = Mul(*_arg, evaluate=True)
+            return TransferFunction.from_rational_expression(res, self.var)
+
+        def _MIMO_doit():
+            _arg = (arg.doit()._expr_mat for arg in reversed(self.args))
+            res = MatMul(*_arg, evaluate=True)
+            return TransferFunctionMatrix.from_Matrix(res, self.var)
+
+        if self.is_SISO:
+            return _SISO_doit()
+        else:
+            return _MIMO_doit()
 
     def _eval_rewrite_as_TransferFunction(self, *args, **kwargs):
+        if not self.is_SISO:
+            raise ValueError("Only transfer functions or a collection of transfer functions"
+                " is allowed in the arguments.")
+        return self.doit()
+
+    def _eval_rewrite_as_TransferFunctionMatrix(self, *args, **kwargs):
+        if self.is_SISO:
+            raise ValueError("Only transfer function matrices or a collection of transfer function"
+                " matrices is allowed in the arguments.")
         return self.doit()
 
     def __add__(self, other):
-        if isinstance(other, (TransferFunction, Series)):
-            if not self.var == other.var:
-                raise ValueError("All the transfer functions should use the same complex variable "
-                    "of the Laplace transform.")
+        try:
+            if self.is_SISO != other.is_SISO:
+                raise TypeError("Cannot add SISO and MIMO systems together.")
+        except AttributeError:
+            raise ValueError(f"Cannot add Series type with {type(other)}.")
```

Comment:
Don't catch `AttributeError`. An `__add__` method should not raise if the argument is not of the expected type. Rather it should return `NotImplemented` which will lead to `TypeError` if the other object also returns `NotImplemented`.